open-api/rest-catalog-open-api.yaml

```diff
@@ -1435,12 +1463,66 @@ components:
           additionalProperties:
             type: string
 
+    TokenRequest:
+      description:
+        OAuth2 client credentials or token exchange request
+
+
+        See -
+
+        - Client credentials flow - https://datatracker.ietf.org/doc/html/rfc6749#section-4.4
+
+        - Token exchange flow - https://datatracker.ietf.org/doc/html/rfc8693
+      type: object
+      required:
+        - grant_type
+      properties:
+        grant_type:
+          type: string
+          enum:
+            - client_credentials
+            - urn:ietf:params:oauth:grant-type:token-exchange
+        client_id:
+          type: string
+        client_secret:
+          type: string
+        subject_token:
```

Comment:
Just to note, the OAuth2 spec [recommends](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1) passing client ID and secret in an `Authorization: Basic...` header, but passing in the request body is supported.

Reply:
I'll add this to the description. I added them here because I saw that they can be passed here. They're also optional. Documenting this route isn't necessarily required, but the Swagger/OpenAPI docs say that you _can_ document the token endpoint and I thought it would be good to do as much as we can.

Reply:
Added.